smart_card_connector_app/src/create_human_readable_supported_readers.py

```diff
@@ -0,0 +1,132 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Creates a human readable file with the list of readers supported by the 
+Connector App."""
+
+import argparse
+import string
+import sys
+
+# List of section titles that should be picked up for forming the USB devices
+# list in the file.
+CCID_SUPPORTED_READERS_CONFIG_SECTIONS_TO_PICK = ('supported', 'shouldwork')
+
+def load_usb_devices(ccid_supported_readers_file):
+  """Parses the specified config containing descriptions of the readers
+  supported by CCID.
+
+  Only readers from the sections listed in the
+  CCID_SUPPORTED_READERS_CONFIG_SECTIONS_TO_PICK are picked up.
+
+  Args:
+    ccid_supported_readers_file: The opened config file containing descriptions
+        of the readers supported by CCID.
+
+  Returns list of strings: (name), which are the names of the usb devices.
+  """
+
+  # The CCID supported readers config file format is the following.
+  #
+  # There are several sections, each of which starts with a comment block
+  # containing the title of the section (e.g. "supported"), followed by a number
+  # of lines containing reader descriptions.
+  #
+  # The reader description line has the following format:
+  # "vendor_id:product_id:name", where vendor_id and product_id are hexadecimal
+  # integers.
+  #
+  # Empty lines should be ignored.
+  #
+  # Comment lines, which start from the hash character, should also be ignored
+  # with the only exception of special-formatted comments containing section
+  # titles.
+
+  # Prefix marker used for determining section titles.
+  SECTION_HEADER_PREFIX = '# section:'
+  # Prefix marker used for determining comment lines.
+  COMMENT_PREFIX = '#'
+
+  usb_devices = []
+  current_section = None
+  ignored_usb_device_count = 0
+  for line in ccid_supported_readers_file:
+    if not line.strip():
+      # Ignore empty line
+      continue
+    if line.startswith(SECTION_HEADER_PREFIX):
+      # Parse section title line
+      current_section = line[len(SECTION_HEADER_PREFIX):].strip()
+      if not current_section:
+        raise RuntimeError('Failed to extract section title from the CCID '
+                           'supported readers config')
+      continue
+    if line.startswith(COMMENT_PREFIX):
+      # Ignore comment line
+      continue
+    # Parse reader description line
+    parts = line.split(':', 2)
+    if len(parts) != 3:
+      raise RuntimeError('Failed to parse the reader description from the CCID '
+                         'supported readers config: "{0}"'.format(line))
+    name = parts[2]
```

Comment:
nit: I think we don't need an empty line between each two entries in the resulting file. So I think here name should not end with a \n (probably best to just strip it from the line before splitting the line).